keel/api/views.py

```diff
@@ -1,3 +1,38 @@
-from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from .permissions import CustomLeadPermission
+from keel.leads.models import CustomerLead
+from keel.leads.serializers import CustomerLeadSerializer
 
-# Create your views here.
+class CustomerLeadView(APIView):
+
+    permission_classes = (CustomLeadPermission, )
+
+    def post(self, request, format="json"):
+        serializer = CustomerLeadSerializer(data=request.data)
+
+        if not serializer.is_valid():
```

Comment:
you can avoid this manual step if you pass the raise exception to the serialiser itself and can get valid data in next step
  serializer.is_valid(raise_exception=True)        
  valid_data = serializer.validated_data

Reply:
done

Reply:
updated please check